forte/processors/base/data_augment_processor.py

```diff
@@ -46,44 +55,559 @@ class ReplacementDataAugmentProcessor(BaseDataAugmentProcessor):
+        called after processing a multipack.
+        """
+        self.replaced_annos.clear()
+        self.inserted_annos.clear()
+        self.deleted_annos.clear()
+        self.data_pack_map.clear()
+        self.anno_maps.clear()
+
+    def _copy_multi_pack_links(self, multi_pack: MultiPack):
+        r"""
+        This function copies the MultiPackLinks in the multipack.
+        It could be used in tasks such as text generation, where
+        MultiPackLink is used to align the source and target.
+
+        Args:
+            - input: The input multi pack.
         """
-        pass
+        for mpl in multi_pack.get(MultiPackLink):
+            # Get the original Entry and DataPack.
+            parent: Entry = mpl.get_parent()
+            child: Entry = mpl.get_child()
+            parent_pack: DataPack = parent.pack
+            child_pack: DataPack = child.pack
+            parent_pack_pid: int = parent_pack.meta.pack_id
+            child_pack_pid: int = child_pack.meta.pack_id
+
+            # Only copy the link when both packs are present.
+            if parent_pack_pid not in self.data_pack_map \
+                    or child_pack_pid not in self.data_pack_map \
+                    or parent_pack_pid not in self.anno_maps \
+                    or child_pack_pid not in self.anno_maps:
+                continue
+            # Get the new Entry and DataPack.
+            new_parent_pack: DataPack = multi_pack.get_pack_at(
+                multi_pack.get_pack_index(self.data_pack_map[parent_pack_pid])
+            )
+            new_child_pack: DataPack = multi_pack.get_pack_at(
+                multi_pack.get_pack_index(self.data_pack_map[child_pack_pid])
+            )
+
+            # Only copy the link when both entries are present.
+            if parent.tid not in self.anno_maps[parent_pack_pid] \
+                    or child.tid not in self.anno_maps[child_pack_pid]:
+                continue
+            new_parent_tid: int = self.anno_maps[parent_pack_pid][parent.tid]
+            new_child_tid: int = self.anno_maps[child_pack_pid][child.tid]
+            new_parent: Entry = new_parent_pack.get_entry(new_parent_tid)
+            new_child: Entry = new_child_pack.get_entry(new_child_tid)
+            # Copy the MultiPackLink.
+            multi_pack.add_entry(
+                MultiPackLink(
+                    multi_pack, new_parent, new_child
+                )
+            )
+        self._clear_states()
+
+    def _process(self, input_pack: MultiPack):
+        replacement_op = create_class_with_kwargs(
+            self.configs["kwargs"]["data_aug_op"],
+            class_args={
+                "configs": self.configs["kwargs"]["data_aug_op_config"]
+            }
+        )
+        augment_entry = get_class(self.configs["augment_entry"])
+        new_packs: List[Tuple[str, DataPack]] = []
+
+        for pack_name, data_pack in input_pack.iter_packs():
+            for anno in data_pack.get(augment_entry):
```

Comment:
I don't see you calling `insert` and `delete` from this class, where are they called?

Reply:
The insert and delete are actually for the EDA. The EDA processors will call the functions in their _process methods. Maybe it would be better to make them private methods, like _insert and _delete.

Reply:
I see. I thought the users need to use them. Then we should make the private.